instrumentation/mongodb-reactive-streams-4.2/src/main/java/com/nr/agent/mongo/MongoUtil.java

```diff
@@ -0,0 +1,156 @@
+
+    public static final String OP_DROP_INDEX = "dropIndex";
+    public static final String OP_DROP_INDEXES = "dropIndexes";
+    public static final String OP_DROP_COLLECTION = "drop";
+    public static final String OP_DROP_DATABASE = "dropDatabase";
+    public static final String OP_DISTINCT = "distinct";
+    public static final String OP_COUNT = "count";
+    public static final String OP_MAP_REDUCE = "mapReduce";
+    public static final String OP_REPLACE_ONE = "replaceOne";
+    public static final String OP_LIST_INDEXES = "listIndexes";
+    public static final String OP_BULK_WRITE = "bulkWrite";
+    public static final String OP_INSERT_ONE = "insertOne";
+    public static final String OP_INSERT_MANY = "insertMany";
+    public static final String OP_UPDATE_MANY = "updateMany";
+    public static final String OP_UPDATE_ONE = "updateOne";
+
+    public static final String OP_DELETE_ONE = "deleteOne";
+    public static final String OP_DELETE_MANY = "deleteMany";
+    public static final String CUSTOM = "Custom";
+
+    public static final String OP_DEFAULT = "other";
+
+    public static final String UNKNOWN = "unknown";
+    public static final String CLUSTER = "Cluster";
+    public static final int DEFAULT_PORT = 27017;
+
+    public static void addDatabaseAndHostToMap(final String databaseName, final String host) {
+        NewRelic.getAgent().getLogger().log(Level.FINE, "Adding mongo DB with with host to map: {0} --> {1}", databaseName, host);
+        if (databaseName != null && host != null) {
+            if (mongoDatabaseToHostMap.containsKey(databaseName)) {
+                mongoDatabaseToHostMap.replace(databaseName, host);
+            } else {
+                mongoDatabaseToHostMap.put(databaseName, host);
+            }
+        }
+    }
+
+    public static String getHostBasedOnDatabaseName(String databaseName) {
+        return mongoDatabaseToHostMap.getOrDefault(databaseName, UNKNOWN);
+    }
+
+
+    public static String determineHostDisplayValueFromCluster(ClusterDescription clusterDescription) {
+        String hostDescription = UNKNOWN;
+        List<ServerDescription> serverDescriptions = clusterDescription.getServerDescriptions();
+
+        if (serverDescriptions != null) {
+            ServerAddress serverAddress = serverDescriptions.get(0).getAddress();
+            hostDescription = serverDescriptions.size() == 1 ?  serverAddress.getHost() + ":" + serverAddress.getPort() : CLUSTER;
+        }
+
+        return hostDescription;
+    }
+
+    public static <T> SingleResultCallback<T> instrumentSingleResultCallback(SingleResultCallback<T> callback, String collectionName,
+            String operationName, String databaseName, String host) {
+        if (callback instanceof NRCallbackWrapper) {
+            return callback;
+        }
+
+        String hostname = null;
+        int port = DEFAULT_PORT;
+        String [] hostAndPort = host.split(":");
```

Comment:
This split call will compile a Pattern everytime it is called.
To improve performance, it is better to do a static variable and use that to split here.

```
    private static final Pattern COLON = Pattern.compile(":");
    
    public static ... instrumentSingleResultCallback(...) {
      String[] hostAndPort = COLON.split(host);
    
    }
```